apps/frontend/src/components/technique/TechniqueTable.tsx

```diff
@@ -0,0 +1,235 @@
+  },
+  {
+    title: 'Instruments',
+    render: (data) => data.instruments.length,
+  },
+];
+
+const TechniqueTable = () => {
+  const {
+    loadingTechniques,
+    techniques,
+    setTechniquesWithLoading: setTechniques,
+  } = useTechniquesData();
+
+  const { api } = useDataApiWithFeedback();
+  const isUserOfficer = useCheckAccess([UserRole.USER_OFFICER]);
+  const [urlQueryParams, setUrlQueryParams] =
+    useQueryParams<UrlQueryParamsType>(DefaultQueryParams);
+
+  const [openTechniqueAssignment, setOpenTechniqueAssignment] = useState(false);
+  const [selectedTechnique, setSelectedTechnique] =
+    useState<TechniqueFragment | null>(null);
+
+  const onTechniqueDelete = async (techniqueDeletedId: number | string) => {
+    try {
+      await api({
+        toastSuccessMessage: 'Technique deleted successfully!',
+      }).deleteTechnique({
+        id: techniqueDeletedId as number,
+      });
+
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  const AssignmentIndIcon = (): JSX.Element => <AssignmentInd />;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function setAssigningTechniqueId(technique: TechniqueFragment): void {
+    setSelectedTechnique(technique);
+    setOpenTechniqueAssignment(true);
+  }
+
+  const assignInstrumentsToTechniques = async (
+    instruments: InstrumentFragment[]
+  ): Promise<void> => {
+    if (selectedTechnique) {
+      const techniqueId = selectedTechnique.id;
+      if (instruments?.length) {
+        await api({
+          toastSuccessMessage: `Instrument/s assigned to the selected technique successfully!`,
+        })
+          .assignInstrumentsToTechnique({
+            instrumentIds: instruments.map((instrument) => instrument.id),
+            techniqueId,
+          })
+          .then(() => {
+            setTechniques((techniques) =>
+              techniques.map((techniqueItem) => {
+                if (techniqueItem.id === techniqueId) {
+                  return {
+                    ...techniqueItem,
+                    instruments: [...techniqueItem.instruments, ...instruments],
+                  };
+                } else {
+                  return techniqueItem;
+                }
+              })
+            );
+          });
+      }
+    }
+  };
+
+  const AssignedInstruments = React.useCallback(({ rowData }) => {
+    return <AssignedInstrumentsTable technique={rowData} />;
+  }, []);
+
+  const removeIntrumentsFromTechnique = async (
+    instrumentIds: number[]
+  ): Promise<void> => {
+    if (selectedTechnique) {
+      const techniqueId = selectedTechnique.id;
+      if (instrumentIds?.length) {
+        instrumentIds.forEach(async (instrumentId) => {
+          await api({
+            toastSuccessMessage: `Instrument/s unassigned from selected technique successfully!`,
+          })
+            .removeInstrumentFromTechnique({
+              instrumentId,
+              techniqueId,
+            })
```

Comment:
Do we need to do multiple backend calls or we can potentially move this to the backend and  let it handle the load and we can update the frontend.